### PR TITLE
pythonPackages.durus: init at 4.2

### PR DESCRIPTION
**Pull request by grindhold**
###### Description of changes

The durus python object database.

###### Things done

- Built on platform(s)
  - [x ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/durus/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    sha256 = "sha256:1gzxg43zawwgqjrfixvcrilwpikb1ix9b7710rsl5ffk7q50yi3c";
+  };
+
+  nativeBuildInputs = [ ];
+  buildInputs = [  ];
+
+  # Checks disabled due to the fact that this needs a
+  # python unittest framework called 'sancho' which is not
+  # covered in nixpkgs yet.
+  doCheck = false;
```
[the remark this answers is not in the document]
Comment (grindhold) [reply pkgs/development/python-modules/durus/default.nix:18]: done

**Comment by sternenseemann** [comment]
@ofborg build python38Packages.durus python3Packages.durus python310Packages.durus

**Review by sternenseemann — APPROVED** [review]

File: pkgs/development/python-modules/durus/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  pythonImportsCheck = [
+    "durus.connection"
+    "durus.file_storage"
+    "durus.client_storage"
+    "durus.sqlite_storage"
+  ];
+
+  meta = with lib; {
+    description = "A python object persistence layer";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/durus/default.nix:28]:
```suggestion
    description = "Object persistence layer";
```

File: pkgs/development/python-modules/durus/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  version = "4.2";
+
+  src = fetchPypi {
+    inherit version pname;
+    sha256 = "sha256:1gzxg43zawwgqjrfixvcrilwpikb1ix9b7710rsl5ffk7q50yi3c";
+  };
+
+  nativeBuildInputs = [ ];
+  buildInputs = [  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/durus/default.nix:14]:
```suggestion
```

File: pkgs/development/python-modules/durus/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  # Checks disabled due to the fact that this needs a
+  # python unittest framework called 'sancho' which is not
+  # covered in nixpkgs yet.
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/durus/default.nix:17]:
```suggestion
  # Checks disabled due to missing python unittest framework 'sancho' in nixpkgs
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 170642` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python310Packages.durus</li>
<li>python39Packages.durus</li>
</ul>
</details>

**Pull request merged**